packages/react-components/src/components/utils/GridLayoutUtils.ts

```diff
@@ -0,0 +1,158 @@
+  }
+  const aspectRatio = width / height;
+  // Approximate how many rows to divide the grid to achieve cells close to the TARGET_CELL_RATIO
+  let rows = Math.floor(Math.sqrt((TARGET_CELL_RATIO / aspectRatio) * numberOfItems)) ?? 1;
+  // Make sure rows do not exceed numberOfItems
+  rows = Math.min(rows, numberOfItems);
+  // Given the rows, get the minimum columns needed to create enough cells for the number of items
+  let columns = Math.ceil(numberOfItems / rows);
+
+  // Default horizontalFill to true
+  let horizontalFill = true;
+
+  while (rows < numberOfItems) {
+    // If cell ratio is less than MINIMUM_CELL_RATIO_ALLOWED then try more rows
+    if ((rows / columns) * aspectRatio >= MINIMUM_CELL_RATIO_ALLOWED) {
+      // If number of items is less than the total cells, we need to figure out whether the big cells should stretch horizontally or vertically
+      // to fill in the empty spaces
+      // e.g. For 2 rows, 3 columns, but only 5 items, we need to choose whether to stetch cells
+      //       horizontally            or           vertically
+      //  ______________________               _______________________
+      // |       |       |      |             |       |       |       |
+      // |_______|_______|______|             |_______|_______|       |
+      // |           |          |             |       |       |       |
+      // |___________|__________|             |_______|_______|_______|
+      if (numberOfItems < rows * columns) {
+        // Calculate the width-to-height ratio of big cells stretched horizontally
+        const horizontallyStretchedCellRatio = (rows / (columns - 1)) * aspectRatio;
+        // Calculate the width-to-height ratio of big cells stretched vertically
+        const verticallyStretchedCellRatio = ((rows - 1) / columns) * aspectRatio;
+        // We know the horizontally stretched cells is higher than MINIMUM_CELL_RATIO_ALLOWED. If vertically stretched cells is also higher than
+        // the MINIMUM_CELL_RATIO_ALLOWED, then choose which ratio is better.
+        if (verticallyStretchedCellRatio >= MINIMUM_CELL_RATIO_ALLOWED) {
+          // If vertically stetched cell has a ratio closer to TARGET_CELL_RATIO then change the flow to vertical
+          if (isCloserThan(verticallyStretchedCellRatio, horizontallyStretchedCellRatio, TARGET_CELL_RATIO)) {
+            horizontalFill = false;
+          }
+        }
+      }
+      break;
+    }
+    rows += 1;
+    columns = Math.ceil(numberOfItems / rows);
+  }
+
+  return { horizontalFill, rows, columns };
+};
+
+/**
+ * Creates a styles classname with CSS Grid related styles given GridProps and the number of items to distribute as evenly as possible.
+ * @param numberOfItems - number of items to place in grid
+ * @param gridProps - GridProps that define the number of rows, number of columns, and the fill direction
+ * @returns
+ */
+export const createGridStyles = (numberOfItems: number, gridProps: GridProps): string => {

```

Comment:
Can we try to split the children into two slices and pass the children down into two block Components.

This might avoid:
* Having to split into finer grid
* Having to style using nth-last-child